packages/web-client/app/components/card-pay/withdrawal-workflow/transaction-approval/index.ts

```diff
@@ -1,17 +1,84 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { toBN } from 'web3-utils';
+import Layer2Network from '@cardstack/web-client/services/layer2-network';
+import {
+  ConvertibleSymbol,
+  TokenDisplayInfo,
+  TokenSymbol,
+} from '@cardstack/web-client/utils/token';
 import { WorkflowCardComponentArgs } from '@cardstack/web-client/models/workflow/workflow-card';
 
 class CardPayWithdrawalWorkflowChooseBalanceComponent extends Component<WorkflowCardComponentArgs> {
-  constructor(owner: unknown, args: WorkflowCardComponentArgs) {
-    super(owner, args);
+  defaultTokenSymbol: TokenSymbol = 'DAI.CPXD';
+  cardTokenSymbol: TokenSymbol = 'CARD.CPXD';
+  @service declare layer2Network: Layer2Network;
+  @tracked tokenSymbol: TokenSymbol =
+    this.args.workflowSession.state.withdrawalToken ?? this.defaultTokenSymbol;
+  @tracked isConfirmed = false;
+
+  get withdrawalToken() {
+    return new TokenDisplayInfo(this.tokenSymbol);
+  }
+
+  get convertibleSymbol(): ConvertibleSymbol {
+    if (this.tokenSymbol === this.cardTokenSymbol) {
+      return 'CARD';
+    } else {
+      return 'DAI';
+    }
+  }
+
+  get tokenBalance() {
+    if (this.tokenSymbol === this.defaultTokenSymbol) {
+      return this.layer2Network.defaultTokenBalance ?? toBN('0');
+    } else if (this.tokenSymbol === this.cardTokenSymbol) {
+      return this.layer2Network.cardBalance ?? toBN('0');
+    }
+    return toBN('0');
+  }
+
+  get withdrawalAmount() {
+    // TODO: Replace with the amount selected in previous card
+    return this.tokenBalance;
+  }
+
+  get depotAddress() {
+    return this.layer2Network.depotSafe?.address || undefined;
+  }
+
+  get isDisabled() {
+    return (
+      !this.depotAddress ||
+      !this.withdrawalAmount ||
+      this.withdrawalAmount.isZero() ||
```

Comment:
Maybe it doesn’t matter but it should be impossible to get to this step with a zero withdrawal amount because of validation in the amount card 🤔